File: essclient/src/stores/EmployeeStore/Employee.js
import { decorate, observable, computed } from 'mobx'

class Employee  {

    constructor(json={}) {
        this.id = json.id || '<New>'
        this.firstName = json.firstName || ''
        this.middleName = json.middleName || ''
        this.lastName = json.lastName || ''
        this.dob = json.dob || ''
        this.address = json.address || ''
        this.email = json.email || ''
        this.phone = json.phone || ''
        this.kinName = json.kinName || ''
        this.kinRelation = json.kinRelation || ''
        this.kinPhone = json.kinPhone || ''
        this.superFund=json.superFund || ''
        this.superAbn = json.superAbn || ''
        this.superNo = json.superNo || ''
        this.bsb = json.bsb || ''
        this.accountNo = json.accountNo || ''
        console.log(this)
    }

    get name() {
        return this.firstName + this.lastName
    }

    update (json) {
        Object.keys(this).forEach ( k => this[k] = json[k])
    }

    isValid () {
        let errors = []
        let warnings = []

        this.firstName.length === 0 && errors.push('First name cannot be empty')
        this.lastName.length === 0 && errors.push('Last name cannot be empty')
        this.dob.length === 0 && errors.push('You must provide your date of birth')
        this.address.length === 0 && errors.push('You must provide your date of address')
        this.email.length === 0 && errors.push('You must provide a valid email address')
        this.phone.length === 0 && errors.push('You must provide a valid phone number')

        this.superFund.length === 0 && warnings.push('Your superannuation payments made to Australian Super if you do not nominate a superannuation fund')
        this.superAbn.length === 0 && warnings.push('Your superannuation payments made to Australian Super if you do not nominate a superannuation fund')
        this.superNo.length === 0 && warnings.push('Your superannuation payments made to Australian Super if you do not nominate a superannuation fund')

        this.bsb.length === 0  && errors.push ('Invalid bsb')
        this.accountNo.length === 0  && errors.push ('Invalid bsb')

    }

}

decorate(Employee, {
    firstName: observable,
    middleName: observable,
    lastName: observable,
    address: observable,
    email: observable,
    phone: observable,
    bsb: observable,
    accountNo: observable,
    name: computed,
})


export default Employee

File: essserver/db/employee.py
from google.appengine.ext import ndb


class Employee(ndb.Model):
    firstName = ndb.StringProperty()
    middleName = ndb.StringProperty()
    lastName = ndb.StringProperty()
    dob = ndb.StringProperty()
    email = ndb.StringProperty()
    address = ndb.StringProperty()
    phone = ndb.StringProperty()
    kinName = ndb.StringProperty()
    kinRelation = ndb.StringProperty()
    kinPhone = ndb.StringProperty()
    superFund=ndb.StringProperty()
    superAbn = ndb.StringProperty()
    superNo = ndb.StringProperty()
    bsb = ndb.StringProperty()
    accountNo = ndb.StringProperty()

    @property
    def serialize(self):
       """Return object data in easily serializeable format"""
       e = self.to_dict()
       e['id'] = self.key.id()
       return e

    @staticmethod
    def save(details):

        if details['id'] == '<New>':
            eId = details['firstName'][0].upper() + details['lastName'][0].upper() + '001'
            e = Employee(key = ndb.Key(Employee, eId))
        else:
            e = Employee.get_by_id(details['id'])


        e.firstName = details['firstName']
        e.middleName = details['middleName']
        e.lastName = details['lastName']
        e.dob = details['dob']
        e.email = details['email']
        e.address = details['address']
        e.phone = details['phone']
        e.kinName = details['kinName']
        e.kinRelation = details['kinRelation']
        e.kinPhone = details['kinPhone']
        e.superFund=details['superFund']
        e.superAbn = details['superAbn']
        e.superNo = details['superNo']
        e.bsb = details['bsb']
        e.accountNo = details['accountNo']

        e.put()

        return e

    @staticmethod
    def get (details):
        if details['id'] == '<New>':
            eId = details['firstName'][0].upper() + details['lastName'][0].upper() + '001'
            eKey = ndb.Key(Employee, eId)
            return Employee(
                key = eKey,
                firstName = details['firstName'],
                lastName = details['lastName'],
                email = details['email'],
                address = details['address'],
                phone = details['phone'],
                bsb = details['bsb'],
                accountNo = details['accountNo'],
            )
        else:
            return Employee.get_by_id(details['id'])